gitlab/resource_gitlab_instance_cluster_test.go

```diff
@@ -0,0 +1,301 @@
+		if cluster.Name != want.Name {
+			return fmt.Errorf("got name %q; want %q", cluster.Name, want.Name)
+		}
+
+		if cluster.Domain != want.Domain {
+			return fmt.Errorf("got domain %q; want %q", cluster.Domain, want.Domain)
+		}
+
+		if cluster.EnvironmentScope != want.EnvironmentScope {
+			return fmt.Errorf("got environment scope %q; want %q", cluster.EnvironmentScope, want.EnvironmentScope)
+		}
+
+		if cluster.PlatformKubernetes.APIURL != want.KubernetesApiURL {
+			return fmt.Errorf("got kubernetes api url %q; want %q", cluster.PlatformKubernetes.APIURL, want.KubernetesApiURL)
+		}
+
+		if cluster.PlatformKubernetes.CaCert != want.KubernetesCACert {
+			return fmt.Errorf("got kubernetes ca cert %q; want %q", cluster.PlatformKubernetes.CaCert, want.KubernetesCACert)
+		}
+
+		if cluster.PlatformKubernetes.AuthorizationType != want.KubernetesAuthorizationType {
+			return fmt.Errorf("got kubernetes authorization type %q; want %q", cluster.PlatformKubernetes.AuthorizationType, want.KubernetesAuthorizationType)
+		}
+
+		return nil
+	}
+}
+
+func testAccGitlabInstanceClusterConfig(rInt int, managed bool) string {
+	m := fmt.Sprintf("%t", managed)
+	if managed {
+		m = "true"
+	}
+
+	return fmt.Sprintf(`
+variable "cert" {
+  default = <<EOF
+%s
+EOF
+}
+
+resource gitlab_instance_cluster "foo" {
+  name                          = "foo-cluster-%d"
+  domain                        = "example.com"
+  managed                       = "%s"
+  kubernetes_api_url            = "https://123.123.123"
+  kubernetes_token              = "some-token"
+  kubernetes_ca_cert            = "${trimspace(var.cert)}"
+  kubernetes_authorization_type = "abac"
+}
+`, instanceClusterFakeCert, rInt, m)
+}
+
+func testAccGitlabInstanceClusterUpdateConfig(rInt int, authType string) string {
+	return fmt.Sprintf(`
+variable "cert" {
+  default = <<EOF
+%s
+EOF
+}
+
+resource gitlab_instance_cluster "foo" {
+  name                          = "foo-cluster-%d"
+  domain                        = "example-new.com"
+  kubernetes_api_url            = "https://124.124.124"
+  kubernetes_token              = "some-token"
+  kubernetes_ca_cert            = "${trimspace(var.cert)}"
+  kubernetes_authorization_type = "%s"
+}
+`, instanceClusterFakeCert, rInt, authType)
+}
+
+func testAccGitlabInstanceClusterManagement(rInt int, managed bool) string {
+	m := "false"
+	if managed {
+		m = "true"
+	}
+
+	return fmt.Sprintf(`
+variable "cert" {
+  default = <<EOF
+%s
+EOF
+}
+
+
+resource gitlab_instance_cluster "foo" {
+  name                          = "foo-cluster-%d"
+  domain                        = "example.com"
+  managed                       = "%s"
+  kubernetes_api_url            = "https://123.123.123"
+  kubernetes_token              = "some-token"
+  kubernetes_ca_cert            = "${trimspace(var.cert)}"
+  kubernetes_authorization_type = "abac"
+
+}
+`, instanceClusterFakeCert, rInt, m)
+}
+
+var instanceClusterFakeCert = `-----BEGIN CERTIFICATE-----
```

Comment:
If the cert expires will it break the test? If so please add a comment showing how it is generated.

Reply:
Good question Adam, I looked into this a little. `instanceClusterFakeCert` has to be a valid PEM certificate, but it doesn't need to be linked to a real cluster. For example I googled "fake pem certificate" and copied [this fake cert](https://gist.github.com/djsauble/f7b04dc331542e7c4419) into `resource_gitlab_instance_cluster_test.go` and the tests ran fine.

Reply:
Cool thanks for checking